### DISCUSSION TITLE
None of the Arduino-IRremote examples work on a LGT8F328P without activating the Serial.begin()

### DISCUSSION THREAD

**Discussion by AKBaltic** [dbuezas/lgt8fx#194] [Q&A]
In discussion on https://github.com/Arduino-IRremote/Arduino-IRremote/discussions/943 , I have already asked a question none of the  Arduino-IRremote examples work on a LGT8F328 without activating the Serial.begin(). Only I don’t understand if this is a problem with Arduino-IRremote libraries or with lgt8fx libraries. 
Тhe question is as follows: none of the Arduino-IRremote receive examples work on a LGT8F328 without activating the Serial.begin(). Even the LED doesn't flash.
// Serial.begin(115200); - does not work.
Serial.begin(115200); - it works.
On the Arduino Nano, everything works with and without the Serial.
Мaybe because of the difference in 16 MHz and 32 MHz. Somewhere it is necessary to change the prescaler, and where I do not know. I ask for help.

**Comment by LaZsolt** [comment]
@AKBaltic 
The `Serial.begin()` function initializes the UART hardware, but the bootloader also set it to some speed, stopbits and parity before the setup section. Arduino Nano's and LGT8F328P MiniEVB's bootloader does not set the same speed. This is the main reason why not working your sketch. I never tried, but I think LGT8F328P MiniEVB board's serial speed, without setting it, is 57600.

So the UART hardware must always be set with the `Serial.begin()` in the setup section, regardless what type of MCU you use.

**Comment by AKBaltic** [comment]
Thank you. Most likely you are right. I also tried this sketch (https://github.com/sui77/rc-switch/tree/master/examples/ReceiveDemo_Simple). The situation is the same:  doesn't work without activation the Serial.begin(). How then to be? The Serial.begin() takes a lot of memory and machine time to execute. 

![LGT8F328-2](https://user-images.githubusercontent.com/97590166/149197327-e642dca4-f54e-4e75-b003-a8b1e5533a08.png)
